### ISSUE TITLE
Performance: Pre-compile minimatch patterns

### ISSUE THREAD

**Issue by jrajav** [eslint/eslint#1095]
minimatch has a method `makeRe` that will precompile a pattern into a regexp. Using this should greatly increase ignored-paths performance, especially on large numbers of patterns and/or file paths. It would also eliminate any overhead from comment patterns. I have not yet tested any performance difference - will do so sometime soon.

**Comment by gyandeeps** [comment]
Started working on this but getting blocked by related bug https://github.com/isaacs/minimatch/issues/29

Everything passes in our test except one,
 here is whats happening:

```js
var filePath = "c:/test/abc.js";
var pattern = "c:/test/abc.js";

minimatch(filepath, pattern) // => match found

filepath.match(minimatch.makeRe(pattern)) // => match not found
```

**Comment by gyandeeps** [comment]
closing in favor of #2365
